vortex-serde/src/layouts/read/cache.rs

```diff
@@ -78,12 +88,17 @@ impl RelativeLayoutCache {
             .remove(&self.absolute_id(path))
     }
 
-    pub fn dtype(&self) -> DType {
-        self.dtype.clone()
+    pub fn has_dtype(&self) -> bool {
+        self.dtype.is_some()
+    }
```

Comment:
A bit of a hack. We know that if we don't have dtype it must be stored somehow. However, I think we can deduce stats schema and avoid storing dtype altogether if we store stats as column